### ISSUE TITLE
Multithreading and CUDA

### ISSUE THREAD

**Issue by jimcodeman** [DanielChappuis/reactphysics3d#151]
Is it possible to add multithreading computation through `std::thread` or CUDA for collision world. Specifically I want to be able to cast rays asynchronously and then wait for the answers or something similar.

**Comment by DanielChappuis** [comment]
Yes it is planned to implement CPU multi-threading. However, I don't have know when exactly. I will release version 0.8.0 of the library in the next days and then focus on robustness. Then I will start working on multi-threading.

**Comment by mrakh** [comment]
User defined callbacks running in parallel might cause undefined behavior if they modify a shared state. In order to guarantee thread safety, you'd have to guard user callback invocations with a mutex, or dispatch callback invocations to a single-reader event queue.
